src/plugins/vis_builder/public/application/components/workspace.tsx

```diff
@@ -42,13 +43,17 @@ export const Workspace: FC = ({ children }) => {
   useEffect(() => {
     async function loadExpression() {
       const schemas = ui.containerConfig.data.schemas;
-      const [valid, errorMsg] = validateSchemaState(schemas, rootState.visualization);
 
-      if (!valid) {
-        if (errorMsg) {
-          toasts.addWarning(errorMsg);
-        }
+      const noAggs = aggConfigs?.aggs?.length === 0;
+      const schemaValidation = validateSchemaState(schemas, rootState.visualization);
+      const aggValidation = validateAggregations(aggConfigs?.aggs || []);
+
+      if (noAggs || !aggValidation.valid || !schemaValidation.valid) {
+        const err = schemaValidation.errorMsg || aggValidation.errorMsg;
```

Comment:
would there be cases where you'd want to show both?

Reply:
Thats a UX question but in my opinion, its not really necessary. There are independent errors and the user can only fix one at a time. And since the app is reactive, as soon as they fix one we will display the next error. The same reactive behavior also makes it quite unlikely that the user can cause more than on error at a time.

Reply:
Yeah, I like that - a comment about the intentional behavior would be nice for future readers.